addon/-legacy-private/system/store.js

```diff
@@ -48,6 +49,8 @@ const badIdFormatAssertion = '`id` passed to `findRecord()` has to be non-empty
 const emberRun = emberRunLoop.backburner;
 const { ENV } = Ember;
 
+const WeakMapImplementation = WeakMap || Ember.WeakMap;
```

Comment:
just use `Ember.WeakMap`. `WeakMap` would error here if undefined and the Ember polyfill uses native if available.